src/custom/hooks/useGetGpApiStatus.ts

```diff
@@ -0,0 +1,39 @@
+import ms from 'ms.macro'
+import { useState, useEffect } from 'react'
+
+type GpQuoteStatus = 'COWSWAP' | 'LEGACY'
+// TODO: use actual API call
+export async function checkGpQuoteApiStatus(): Promise<GpQuoteStatus> {
+  return new Promise((accept) => setTimeout(() => accept('COWSWAP'), 500))
+}
+const GP_QUOTE_STATUS_INTERVAL_TIME = ms`2 hours`
+
+export default function useCheckGpQuoteStatus(defaultApiToUse: GpQuoteStatus): GpQuoteStatus {
+  const [gpQuoteApiStatus, setGpQuoteApiStatus] = useState<GpQuoteStatus>(defaultApiToUse)
+
+  useEffect(() => {
+    console.debug('[useGetQuoteCallback::GP API Status]::', gpQuoteApiStatus)
```

Comment:
What about moving this inside the `checkStatus` function?
This way the `useEffect` doesn't have to depend on `gpQuoteApiStatus` and thus have one less execution when it updates (assuming we start with one type and the first call returns a different type)

Reply:
hmm i don't really understand, maybe it's because of the code selected, can you expand?

Reply:
I meant the `debug` log.

You are logging `gpQuoteApiStatus`, and for that reason the `useEffect` depends on it.

When the inner function `checkStatus` runs, it might update `gpQuoteApiStatus`.
When that happens, it'll trigger the `useEffect` again straight away

Does it make sense?

Reply:
@alfetopito thanks, will address in a separate PR at the end